skyportal/facility_apis/swift.py

```diff
@@ -82,10 +88,142 @@ def _build_payload(self, request):
         return too
 
 
+class XRTAPIRequest:
+
+    """A JSON structure for Swift XRT API requests."""
+
+    def __init__(self, request):
+        """Initialize XRT API request.
+
+        Parameters
+        ----------
+        request : skyportal.models.FollowupRequest
+            The request to add to the queue and the SkyPortal database.
+
+        """
+
+        self.requestgroup = self._build_payload(request)
+
+    def _build_payload(self, request):
+        """Payload json for Swift UVOT/XRT TOO requests.
+
+        Parameters
+        ----------
+        request : skyportal.models.FollowupRequest
+            The request to add to the queue and the SkyPortal database.
+
+        Returns
+        -------
+        payload : swifttools.swift_too.Swift_TOO
+            payload for requests.
+        """
+
+        from swifttools.xrt_prods import XRTProductRequest
+
+        altdata = request.allocation.altdata
+
+        T0 = Time(request.payload["T0"], format='iso')
+        MET = Time('2001-01-01 00:00:00', format='iso')
+        Tdiff = (T0 - MET).jd * 86400
+
+        if "detornot" in request.payload and request.payload["detornot"]:
+            centroid = True
+        else:
+            centroid = False
+
+        myReq = XRTProductRequest(altdata["XRT_UserID"])
+        myReq.setGlobalPars(
+            getTargs=True,
+            centroid=centroid,
+            name=request.obj.id,
+            RA=request.obj.ra,
+            Dec=request.obj.dec,
+            centMeth=request.payload["centMeth"],
+            detMeth=request.payload["detMeth"],
+            useSXPS=False,
+            T0=Tdiff,
+            posErr=request.payload["poserr"],
+        )
+        myReq.addLightCurve(binMeth='counts', pcCounts=20, wtCounts=30, dynamic=True)
+        myReq.addSpectrum(hasRedshift=False)
+        myReq.addStandardPos()
+        myReq.addEnhancedPos()
+        myReq.addAstromPos(useAllObs=True)
+
+        return myReq
+
+
 class UVOTXRTAPI(FollowUpAPI):
 
     """An interface to Swift operations."""
 
+    @staticmethod
+    def get(request, session):
+
+        """Get an analysis request result from Swift.
```

Comment:
analysis or followup request?

Reply:
It is a reduction, so I think it is the right phrasing.